.github/bin/build-matrix-from-impacted.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+
+import argparse
+import yaml
+import json
+import logging
+import sys
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Filter test matrix modules using list of impacted modules."
+    )
+    parser.add_argument(
+        "-m",
+        "--matrix",
+        type=argparse.FileType("r"),
+        default=".github/test-matrix.yaml",
+        help="A YAML file with the test matrix",
+    )
+    parser.add_argument(
+        "-i",
+        "--impacted",
+        type=argparse.FileType("r"),
+        default="gib-impacted.log",
+        help="List of impacted modules, one per line",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        type=argparse.FileType("w"),
+        default=sys.stdout,
+        help="Filename to write JSON output to",
+    )
+    parser.add_argument(
+        "-d",
+        "--debug",
+        action="store_const",
+        dest="loglevel",
+        const=logging.DEBUG,
+        default=logging.WARNING,
+        help="Print lots of debugging statements",
+    )
+
+    args = parser.parse_args()
+    logging.basicConfig(level=args.loglevel)
+    build(args.matrix, args.impacted, args.output)
+
+
+def build(matrix_file, impacted_file, output_file):
```

Comment:
I'd propose (as a follow-up) to move this to a `dev` directory and add some basic tests there too (i.e. handwritten list of modules and impacted modules) to ensure we can confidently change/refactor in future.

Reply:
Sure, I'm still waiting for some confirmation the whole idea is ok.